aten/src/ATen/native/IndexingUtils.h

```diff
@@ -14,14 +14,14 @@ static void invalid_mask(const Tensor & self, int64_t idx, const Tensor & mask,
 }
 
 
-static C10_UNUSED std::vector<Tensor> expandTensors(const Tensor & self, const torch::List<c10::optional<Tensor>>& indices) {
+static C10_UNUSED std::vector<Tensor> expandTensors(const Tensor & self, IOptTensorListRef indices) {
   // If indices come in as ByteTensor or BoolTensor (masks), expand them into the equivalent indexing by LongTensors
   std::vector<Tensor> result;
-  for (c10::optional<Tensor> index_opt : indices) {
+  for (const auto& index_opt : indices) {
     if (!index_opt.has_value()) {
       result.emplace_back();
     } else {
-      Tensor index = std::move(*index_opt);
+      Tensor index = *index_opt;
```

Comment:
Index is not necessarily moved into the result, so it would be profitable to keep it a reference